cat-home/src/test/java/com/dianping/cat/report/task/TaskHelperTest.java

```diff
@@ -65,4 +76,84 @@ public void testTodayZero() {
 		Assert.assertEquals(cal.getTimeInMillis(), todayZero.getTime() + 1L);
 	}
 
+    @Test
+    public void testTodayZero2() {
+        Assert.assertEquals(new Date(1562713200000L),
+                TaskHelper.todayZero(null));
+        Assert.assertEquals(new Date(1562713200000L),
+                TaskHelper.todayZero(new Date()));
+    }
+
+    @Test
+    public void testJoin1() {
+
+        final int[] array = {1, -8, -7, -7, -7, -7, -7, -253_968, -7, -24, -24,
+                -2147475464, 4088, -2147483647, 9, 33505273, 1, -8};
+
+
+        Assert.assertNull(TaskHelper.join((int[]) null, '\u0000',
+                -160729767, -159952269));
+        Assert.assertNull(TaskHelper.join((Number[]) null, '\u0000',
+                -1, 2147483647));
+        Assert.assertNull(TaskHelper.join((double[]) null, '\u0000',
+                -402653175, 1669333001));
+
+        Assert.assertEquals("", TaskHelper.join(new int[0], '\u0000'));
+        Assert.assertEquals("", TaskHelper.join(new double[]{5.1}, '\u0000'));
+        Assert.assertEquals("", TaskHelper.join(new int[0], '\u0000',
+                -2147483648, 0));
+        Assert.assertEquals("", TaskHelper.join(new int[0], '\u0000',
+                1816002434, -1402208264));
+        Assert.assertEquals("", TaskHelper.join(new Number[]{null}, '\u0000',
+                2147483647, 1073742086));
+        Assert.assertEquals("", TaskHelper.join(new double[]{
+                        5.1, 3.5}, '\u0000',
```

Comment:
would probably recommend putting the newline after the `new` instead of after the `{` so that the objects you are comparing are on the same line as this makes it easier to read

Reply:
done

Reply:
@shafi-1 I can't see this as done? It looks like it's still being split on the `{ `

Reply:
should be done now